wp/resources/sample_binaries/Makefile

```diff
@@ -1,6 +1,7 @@
 SAMPLE_DIRS = $(wildcard */)
 BUILD_DIRS = $(SAMPLE_DIRS:%=build-%)
 CLEAN_DIRS = $(SAMPLE_DIRS:%=clean-%)
+MAKEFLAGS += -j$(shell nproc)
 
 .PHONY: all subdirs $(BUILD_DIRS)
```

Comment:
I don't think that the child targets are treated as PHONY. They're cached and not rebuilt every time (as far as I can tell). However, `make` *is* invoked on each of the subdirectories. I've removed the PHONY annotation and the directories still rebuild on change. I'm going to leave it like that since I can't think of a good reason to keep around PHONY.

Reply:
I am also not an expert, but I thought the general rule was "any target whose name is the filename of the thing it builds should be marked PHONY".  It's entirely possible I'm wrong.

Reply:
I seem to remember reading (I believe in the manual) that PHONY targets are not checked/tracked to see if they've been updated. In effect, something marked PHONY runs every time.